File: back-end/HealthCatalystPeople.Data/Entities/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace HealthCatalystPeople.Data.Entities {
    public class Person {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int? Age { get; set; }
        public string Picture { get; set; }
        public string Email { get; set; }
        public DateTime? Created { get; set; }
        public string Notes { get; set; }
        
        [ForeignKey("Occupations")]
        public int? OccupationId { get; set; }
        public Occupation Occupation { get; set; }
        
        [ForeignKey("Addresses")]
        public int? AddressId { get; set; }
        public Address Address { get; set; }

        [NotMapped]
        public List<Interest> Interests { get; set; }
        [NotMapped]
        public List<PersonInterest> PersonInterests { get; set; } = null;
    }
}

File: front-end/src/app/types/person.ts
import { Address } from './address';
import { Occupation } from './occupation';
import { Interest } from './interest';

export interface Person {
    id?: number;
    firstName: string;
    lastName: string;
    address: Address;
    occupation: Occupation;
    interests: Interest[];
    picture: string;
    email?: string;
    age: number;
    notes?: string;
}

export type PersonUpdate = Partial<Person>;
